biogrid/libs/biogrid-simulator/src/lib/biobattery/biobattery-class.ts

```diff
@@ -24,7 +24,7 @@ export class BioBattery implements Battery {
 
   supplyPower(outputenergy: Energy): Energy {
     if (this.currentBatteryPower - outputenergy < 0) {
```

Comment:
Should the battery supply all its available power? E.g. if you request 10 units but the battery only has 5, should the battery at least supply all the available 5 units? This function can then return how much energy was actually supplied, and the caller can figure out if another biobattery should be called for the remainder

Reply:
> Should the battery supply all its available power? E.g. if you request 10 units but the battery only has 5, should the battery at least supply all the available 5 units? This function can then return how much energy was actually supplied, and the caller can figure out if another biobattery should be called for the remainder

That's the end goal. I am still figuring the best way to implement this, and send a message back to the user. I will look into return an interface/type with the message

Reply:
Can you have it return suppliedEnergy of type Energy? I think it would be best to make this change in this PR, but will leave that up to you.